### ISSUE TITLE
Unable to install rmagick 4.0.0 with TruffleRuby 20.0.0

### ISSUE THREAD

**Issue by tiredenzo** [oracle/truffleruby#1954]
Hello,

while trying to

```gem install rmagick -v '4.0.0'```

with

```truffleruby-20.0.0 [ x86_64 ] (rvm osx)```

I get the following stack trace

```➜  vecto git:(master) ✗ gem install rmagick -v '4.0.0'
Building native extensions. This could take a while...
ERROR:  Error installing rmagick:
	ERROR: Failed to build gem native extension.

    current directory: /Users/enzo/.rvm/gems/truffleruby-20.0.0/gems/rmagick-4.0.0/ext/RMagick
/Users/enzo/.rvm/rubies/truffleruby-20.0.0/bin/truffleruby -I /Users/enzo/.rvm/rubies/truffleruby-20.0.0/lib/mri -r ./siteconf20200305-96969-1986otf.rb extconf.rb
checking for brew... yes
checking for /Users/enzo/.rvm/rubies/truffleruby-20.0.0/lib/sulong/native/bin/graalvm-native-clang... yes
checking for pkg-config... yes
checking for outdated ImageMagick version (<= 6.7.7)... no
checking for __GNUC__... yes
checking for Ruby version >= 2.3.0... yes
checking for magick/MagickCore.h... yes
checking for GetImageChannelEntropy() in assert.h,ctype.h,stdio.h,stdlib.h,math.h,time.h,sys/types.h,magick/MagickCore.h... yes
checking for SetImageGray() in assert.h,ctype.h,stdio.h,stdlib.h,math.h,time.h,sys/types.h,magick/MagickCore.h... yes
creating extconf.h
creating Makefile


======================================================================
Thu 05 Mar 20 06:33:49
This installation of RMagick 4.0.0 is configured for
Ruby 2.6.5 (x86_64-darwin18) and ImageMagick 6.9.10
======================================================================


Configured compile options: {:magick_version=>"6.9.10", :local_libs=>" -L/usr/local/Cellar/imagemagick@6/6.9.10-77/lib -lMagickCore-6.Q16", :cflags=>" -DMAGICKCORE_HDRI_ENABLE=0 -DMAGICKCORE_QUANTUM_DEPTH=16 -I/usr/local/Cellar/imagemagick@6/6.9.10-77/include/ImageMagick-6 -std=gnu99", :cppflags=>" -DMAGICKCORE_HDRI_ENABLE=0 -DMAGICKCORE_QUANTUM_DEPTH=16 -I/usr/local/Cellar/imagemagick@6/6.9.10-77/include/ImageMagick-6", :ldflags=>" -L/usr/local/Cellar/imagemagick@6/6.9.10-77/lib -lMagickCore-6.Q16", :defs=>[], :config_h=>"Makefile rmagick.h"}

current directory: /Users/enzo/.rvm/gems/truffleruby-20.0.0/gems/rmagick-4.0.0/ext/RMagick
make "DESTDIR=" clean

current directory: /Users/enzo/.rvm/gems/truffleruby-20.0.0/gems/rmagick-4.0.0/ext/RMagick
make "DESTDIR="
compiling rmagick.c
compiling rmdraw.c
compiling rmenum.c
compiling rmfill.c
compiling rmilist.c
compiling rmimage.c
rmimage.c:12587:1: warning: implicit declaration of function 'C__Bool_to_R__Bool' is invalid in C99 [-Wimplicit-function-declaration]
DEF_ATTR_ACCESSOR(Image, start_loop, bool)
^
./rmagick.h:461:5: note: expanded from macro 'DEF_ATTR_ACCESSOR'
    DEF_ATTR_READER(class, attr, type)\
    ^
./rmagick.h:418:16: note: expanded from macro 'DEF_ATTR_READER'
        return C_##type##_to_R_##type(ptr->attr);\
               ^
<scratch space>:180:1: note: expanded from here
C__Bool_to_R__Bool
^
rmimage.c:12587:1: warning: incompatible integer to pointer conversion returning 'int' from a function with result type 'VALUE' (aka 'void *') [-Wint-conversion]
DEF_ATTR_ACCESSOR(Image, start_loop, bool)
^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./rmagick.h:461:5: note: expanded from macro 'DEF_ATTR_ACCESSOR'
    DEF_ATTR_READER(class, attr, type)\
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./rmagick.h:418:16: note: expanded from macro 'DEF_ATTR_READER'
        return C_##type##_to_R_##type(ptr->attr);\
               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
<scratch space>:180:1: note: expanded from here
C__Bool_to_R__Bool
^
rmimage.c:12587:1: warning: implicit declaration of function 'R__Bool_to_C__Bool' is invalid in C99 [-Wimplicit-function-declaration]
./rmagick.h:462:5: note: expanded from macro 'DEF_ATTR_ACCESSOR'
    DEF_ATTR_WRITER(class, attr, type)
    ^
./rmagick.h:441:21: note: expanded from macro 'DEF_ATTR_WRITER'
        ptr->attr = R_##type##_to_C_##type(val);\
                    ^
<scratch space>:186:1: note: expanded from here
R__Bool_to_C__Bool
^
rmimage.c:14874:21: warning: enumeration value 'StaticGravity' not handled in switch [-Wswitch]
            switch (gravity)
                    ^
4 warnings generated.
compiling rminfo.c
rminfo.c:244:1: warning: implicit declaration of function 'C__Bool_to_R__Bool' is invalid in C99 [-Wimplicit-function-declaration]
DEF_ATTR_ACCESSOR(Info, antialias, bool)
^
./rmagick.h:461:5: note: expanded from macro 'DEF_ATTR_ACCESSOR'
    DEF_ATTR_READER(class, attr, type)\
    ^
./rmagick.h:418:16: note: expanded from macro 'DEF_ATTR_READER'
        return C_##type##_to_R_##type(ptr->attr);\
               ^
<scratch space>:246:1: note: expanded from here
C__Bool_to_R__Bool
^
rminfo.c:244:1: warning: incompatible integer to pointer conversion returning 'int' from a function with result type 'VALUE' (aka 'void *') [-Wint-conversion]
DEF_ATTR_ACCESSOR(Info, antialias, bool)
^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./rmagick.h:461:5: note: expanded from macro 'DEF_ATTR_ACCESSOR'
    DEF_ATTR_READER(class, attr, type)\
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./rmagick.h:418:16: note: expanded from macro 'DEF_ATTR_READER'
        return C_##type##_to_R_##type(ptr->attr);\
               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
<scratch space>:246:1: note: expanded from here
C__Bool_to_R__Bool
^
rminfo.c:244:1: warning: implicit declaration of function 'R__Bool_to_C__Bool' is invalid in C99 [-Wimplicit-function-declaration]
./rmagick.h:462:5: note: expanded from macro 'DEF_ATTR_ACCESSOR'
    DEF_ATTR_WRITER(class, attr, type)
    ^
./rmagick.h:441:21: note: expanded from macro 'DEF_ATTR_WRITER'
        ptr->attr = R_##type##_to_C_##type(val);\
                    ^
<scratch space>:252:1: note: expanded from here
R__Bool_to_C__Bool
^
rminfo.c:1120:1: warning: implicit declaration of function 'C__Bool_to_R__Bool' is invalid in C99 [-Wimplicit-function-declaration]
DEF_ATTR_ACCESSOR(Info, dither, bool)
^
./rmagick.h:461:5: note: expanded from macro 'DEF_ATTR_ACCESSOR'
    DEF_ATTR_READER(class, attr, type)\
    ^
./rmagick.h:418:16: note: expanded from macro 'DEF_ATTR_READER'
        return C_##type##_to_R_##type(ptr->attr);\
               ^
<scratch space>:278:1: note: expanded from here
C__Bool_to_R__Bool
^
rminfo.c:1120:1: warning: incompatible integer to pointer conversion returning 'int' from a function with result type 'VALUE' (aka 'void *') [-Wint-conversion]
DEF_ATTR_ACCESSOR(Info, dither, bool)
^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./rmagick.h:461:5: note: expanded from macro 'DEF_ATTR_ACCESSOR'
    DEF_ATTR_READER(class, attr, type)\
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./rmagick.h:418:16: note: expanded from macro 'DEF_ATTR_READER'
        return C_##type##_to_R_##type(ptr->attr);\
               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
<scratch space>:278:1: note: expanded from here
C__Bool_to_R__Bool
^
rminfo.c:1120:1: warning: implicit declaration of function 'R__Bool_to_C__Bool' is invalid in C99 [-Wimplicit-function-declaration]
./rmagick.h:462:5: note: expanded from macro 'DEF_ATTR_ACCESSOR'
    DEF_ATTR_WRITER(class, attr, type)
    ^
./rmagick.h:441:21: note: expanded from macro 'DEF_ATTR_WRITER'
        ptr->attr = R_##type##_to_C_##type(val);\
                    ^
<scratch space>:284:1: note: expanded from here
R__Bool_to_C__Bool
^
rminfo.c:1745:1: warning: implicit declaration of function 'C__Bool_to_R__Bool' is invalid in C99 [-Wimplicit-function-declaration]
DEF_ATTR_ACCESSOR(Info, monochrome, bool)
^
./rmagick.h:461:5: note: expanded from macro 'DEF_ATTR_ACCESSOR'
    DEF_ATTR_READER(class, attr, type)\
    ^
./rmagick.h:418:16: note: expanded from macro 'DEF_ATTR_READER'
        return C_##type##_to_R_##type(ptr->attr);\
               ^
<scratch space>:25:1: note: expanded from here
C__Bool_to_R__Bool
^
rminfo.c:1745:1: warning: incompatible integer to pointer conversion returning 'int' from a function with result type 'VALUE' (aka 'void *') [-Wint-conversion]
DEF_ATTR_ACCESSOR(Info, monochrome, bool)
^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./rmagick.h:461:5: note: expanded from macro 'DEF_ATTR_ACCESSOR'
    DEF_ATTR_READER(class, attr, type)\
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./rmagick.h:418:16: note: expanded from macro 'DEF_ATTR_READER'
        return C_##type##_to_R_##type(ptr->attr);\
               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
<scratch space>:25:1: note: expanded from here
C__Bool_to_R__Bool
^
rminfo.c:1745:1: warning: implicit declaration of function 'R__Bool_to_C__Bool' is invalid in C99 [-Wimplicit-function-declaration]
./rmagick.h:462:5: note: expanded from macro 'DEF_ATTR_ACCESSOR'
    DEF_ATTR_WRITER(class, attr, type)
    ^
./rmagick.h:441:21: note: expanded from macro 'DEF_ATTR_WRITER'
        ptr->attr = R_##type##_to_C_##type(val);\
                    ^
<scratch space>:31:1: note: expanded from here
R__Bool_to_C__Bool
^
9 warnings generated.
compiling rmkinfo.c
compiling rmmain.c
rmmain.c:1672:15: warning: incompatible integer to pointer conversion passing 'int' to parameter of type 'VALUE' (aka 'void *') [-Wint-conversion]
    if (RTEST(rb_const_defined(rb_cObject, bypass)) && RTEST(rb_const_get(rb_cObject, bypass)))
              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/enzo/.rvm/rubies/truffleruby-20.0.0/lib/cext/include/ruby/ruby.h:482:17: note: passing argument to parameter 'value' here
int RTEST(VALUE value);
                ^
1 warning generated.
compiling rmmontage.c
compiling rmpixel.c
compiling rmstruct.c
rmstruct.c:411:12: error: called object type 'VALUE' (aka 'void *') is not a function or function pointer
    return rb_funcall(Class_Font, rm_ID_new, 9
           ^
/Users/enzo/.rvm/rubies/truffleruby-20.0.0/lib/cext/include/truffleruby/truffleruby.h:146:41: note: expanded from macro 'rb_funcall'
#define rb_funcall(object, method, ...) rb_tr_funcall(object, method, __VA_ARGS__)
                                        ^                             ~~~~~~~~~~~
/Users/enzo/.rvm/rubies/truffleruby-20.0.0/lib/cext/include/truffleruby/truffleruby.h:145:46: note: expanded from macro 'rb_tr_funcall'
#define rb_tr_funcall(object, method, n,...) RUBY_CEXT_INVOKE("rb_funcall", object, method, INT2FIX(n), ##__VA_ARGS__)
                                             ^                                                            ~~~~~~~~~~~
/Users/enzo/.rvm/rubies/truffleruby-20.0.0/lib/cext/include/truffleruby/truffleruby.h:136:37: note: expanded from macro 'RUBY_CEXT_INVOKE'
#define RUBY_CEXT_INVOKE(NAME, ...) RUBY_INVOKE_IMPL(RUBY_CEXT, NAME, ##__VA_ARGS__)
                                    ^                                   ~~~~~~~~~~~
/Users/enzo/.rvm/rubies/truffleruby-20.0.0/lib/cext/include/truffleruby/truffleruby.h:130:54: note: expanded from macro 'RUBY_INVOKE_IMPL'
#define RUBY_INVOKE_IMPL(RECV, NAME, ...) rb_tr_wrap(RUBY_INVOKE_IMPL_NO_WRAP(RECV, NAME, ##__VA_ARGS__))
                                                     ^                                      ~~~~~~~~~~~
/Users/enzo/.rvm/rubies/truffleruby-20.0.0/lib/cext/include/truffleruby/truffleruby.h:129:310: note: expanded from macro 'RUBY_INVOKE_IMPL_NO_WRAP'
#define RUBY_INVOKE_IMPL_NO_WRAP(RECV, NAME, ...) INVOKE_IMPL(RECV, NAME, ##__VA_ARGS__, RUBY_INVOKE_IMPL_10, RUBY_INVOKE_IMPL_9, RUBY_INVOKE_IMPL_8, RUBY_INVOKE_IMPL_7, RUBY_INVOKE_IMPL_6, RUBY_INVOKE_IMPL_5, RUBY_INVOKE_IMPL_4, RUBY_INVOKE_IMPL_3, RUBY_INVOKE_IMPL_2, RUBY_INVOKE_IMPL_1, RUBY_INVOKE_IMPL_0)(RECV, NAME, ##__VA_ARGS__)
                                                                            ~~~~~~~~~~~                                                                                                                                                                                                                              ^
rmstruct.c:865:12: error: called object type 'VALUE' (aka 'void *') is not a function or function pointer
    return rb_funcall(Class_TypeMetric, rm_ID_new, 9
           ^
/Users/enzo/.rvm/rubies/truffleruby-20.0.0/lib/cext/include/truffleruby/truffleruby.h:146:41: note: expanded from macro 'rb_funcall'
#define rb_funcall(object, method, ...) rb_tr_funcall(object, method, __VA_ARGS__)
                                        ^                             ~~~~~~~~~~~
/Users/enzo/.rvm/rubies/truffleruby-20.0.0/lib/cext/include/truffleruby/truffleruby.h:145:46: note: expanded from macro 'rb_tr_funcall'
#define rb_tr_funcall(object, method, n,...) RUBY_CEXT_INVOKE("rb_funcall", object, method, INT2FIX(n), ##__VA_ARGS__)
                                             ^                                                            ~~~~~~~~~~~
/Users/enzo/.rvm/rubies/truffleruby-20.0.0/lib/cext/include/truffleruby/truffleruby.h:136:37: note: expanded from macro 'RUBY_CEXT_INVOKE'
#define RUBY_CEXT_INVOKE(NAME, ...) RUBY_INVOKE_IMPL(RUBY_CEXT, NAME, ##__VA_ARGS__)
                                    ^                                   ~~~~~~~~~~~
/Users/enzo/.rvm/rubies/truffleruby-20.0.0/lib/cext/include/truffleruby/truffleruby.h:130:54: note: expanded from macro 'RUBY_INVOKE_IMPL'
#define RUBY_INVOKE_IMPL(RECV, NAME, ...) rb_tr_wrap(RUBY_INVOKE_IMPL_NO_WRAP(RECV, NAME, ##__VA_ARGS__))
                                                     ^                                      ~~~~~~~~~~~
/Users/enzo/.rvm/rubies/truffleruby-20.0.0/lib/cext/include/truffleruby/truffleruby.h:129:310: note: expanded from macro 'RUBY_INVOKE_IMPL_NO_WRAP'
#define RUBY_INVOKE_IMPL_NO_WRAP(RECV, NAME, ...) INVOKE_IMPL(RECV, NAME, ##__VA_ARGS__, RUBY_INVOKE_IMPL_10, RUBY_INVOKE_IMPL_9, RUBY_INVOKE_IMPL_8, RUBY_INVOKE_IMPL_7, RUBY_INVOKE_IMPL_6, RUBY_INVOKE_IMPL_5, RUBY_INVOKE_IMPL_4, RUBY_INVOKE_IMPL_3, RUBY_INVOKE_IMPL_2, RUBY_INVOKE_IMPL_1, RUBY_INVOKE_IMPL_0)(RECV, NAME, ##__VA_ARGS__)
                                                                            ~~~~~~~~~~~                                                                                                                                                                                                                              ^
2 errors generated.
make: *** [rmstruct.o] Error 1

make failed, exit code 2

Gem files will remain installed in /Users/enzo/.rvm/gems/truffleruby-20.0.0/gems/rmagick-4.0.0 for inspection.
Results logged to /Users/enzo/.rvm/gems/truffleruby-20.0.0/extensions/x86_64-darwin/20.0.0/rmagick-4.0.0/gem_make.out
```


Also an Rmagick issue here [#1882],
Thank you

**Comment by eregon** [comment]
Looks at least part of it is due to `VALUE` being `void*`, I'm working on having `VALUE` as `unsigned long` like MRI.

**Comment by eregon** [comment]
Fixed by 0552ac72e3396492dc42ea53c1ac5ee2eb24a3ba, `gem install rmagick` works now, thank you for reporting the issue.
